File: 프로그래머스/연습문제/2개 이하로 다른 비트.js
const find = (num) => {
	if (num % 2 === 0) return num + 1;

	const num_bin = Array.from("0" + num.toString(2), Number);
	let index = num_bin.length - 1;
	while (true) {
		index--;
		if (num_bin[index] === 0) {
			num_bin[index] = 1;
			num_bin[index + 1] = 0;
			break;
		}
	}

	return parseInt(num_bin.join(""), 2);
};

function solution(numbers) {
	return numbers.map(find);
}

File: 프로그래머스/연습문제/2개 이하로 다른 비트.py
def find(num):
    if not num % 2:
        return num + 1

    num_bin = list(map(int, "0" + format(num, 'b')))
    index = -1
    while True:
        index -= 1
        if not num_bin[index]:
            num_bin[index] = 1
            num_bin[index+1] = 0
            break
    return int(''.join(list(map(str, num_bin))), 2)


def solution(numbers):
    return [find(number) for number in numbers]